packages/vscode-extension/src/debug/prerequisitesHandler.ts

```diff
@@ -369,19 +370,39 @@ async function checkNpmInstall(component: string, folder: string): Promise<Check
   let error = undefined;
   try {
     if (!installed) {
-      VsCodeLogInstance.outputChannel.appendLine(`Npm installing (${component})`);
-      const exitCode = await runTask(
-        new vscode.Task(
-          {
-            type: "shell",
-            command: `${component} npm install`,
-          },
-          vscode.workspace.workspaceFolders![0],
-          `${component} npm install`,
-          ProductName,
-          new vscode.ShellExecution(npmInstallCommand, { cwd: folder })
-        )
-      );
+      let exitCode: number | undefined;
+
+      let running = false;
+      for (const [key, value] of trackedTasks) {
+        if (value === `${component} npm install`) {
+          running = true;
+          break;
+        }
+      }
+      if (running) {
+        exitCode = await new Promise((resolve: (value: number | undefined) => void) => {
+          const endListener = taskEndEventEmitter.event((result) => {
+            if (result.name === `${component} npm install`) {
+              endListener.dispose();
+              resolve(result.exitCode);
+            }
```

Comment:
A very corner case is that task ended just now (between line 378 and line 384)...
Better to check running again here.
```
for (... trackedTasks) ...
if (!running) {
  endListener.dispose();
  resolve(undefined);
}
```

Reply:
Nice catch. Added double check.